share/doc/homebrew/Python-for-Formula-Authors.md

```diff
@@ -54,40 +54,32 @@ Formulæ for apps that require Python 3 **should** declare an unconditional depe
-```
-The first line copies all of the executables to bin. The second line writes stubs to bin that set `PYTHONPATH` and moves the original files back to `libexec/"bin"`.
+Homebrew provides helper methods for instantiating and populating virtualenvs. You can use them by putting `include Language::Python::Virtualenv` on the `Formula` class definition, above `def install`.
 
-## Dependencies
-
-All Python dependencies of applications that are not packaged by Homebrew (and those dependencies' Python dependencies, recursively) **should** be unconditionally downloaded as `Resource`s and installed into the application keg's `libexec/"vendor"` path. This prevents the state of the system Python packages from being affected by installing an app with Homebrew and guarantees that apps use versions of their dependencies that are known to work together. `libexec/"vendor"` is preferred to `libexec` so that formulæ don't accidentally install executables belonging to their dependencies, which can cause linking conflicts.
-
-Each dependency **should** be explicitly installed; please do not rely on setup.py or pip to perform automatic dependency resolution, for the [reasons described here](Acceptable-Formulae.md#we-dont-like-install-scripts-that-download-things).
+An example of using the virtualenv helper methods is:
 
-You can use [homebrew-pypi-poet](https://pypi.python.org/pypi/homebrew-pypi-poet) to help you write resource stanzas. To use it, set up a virtualenv and install your package and all its dependencies. Then, `pip install homebrew-pypi-poet` into the same virtualenv. `poet -f foo` will draft a complete formula for you, or `poet foo` will just generate the resource stanzas.
-
-Set `PYTHONPATH` to include the `libexec/"vendor"` site-packages path with:
-```ruby
-ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
-```
-before staging and installing each resourced dependency with:
 ```ruby
-system "python", *Language::Python.setup_install_args(libexec/"vendor")
+# Create a virtualenv in `libexec`. If your app needs Python 3, make sure that
+# `depends_on :python3` is declared, and use `virtualenv(libexec, "python3")`.
+virtualenv(libexec) do |venv|
+  # Install all of the resources declared on the formula into the virtualenv.
+  venv.pip_install resources
+  # `link_scripts` takes a look at the virtualenv's bin directory before and
+  # after executing the block which is passed into it. If the block caused any
+  # new scripts to be written to the virtualenv's bin directory, link_scripts
+  # will symlink those scripts into the path given as its argument (here, the
+  # formula's `bin` directory in the Cellar.)
+  # pip_install "." will install the package that the formula points to, because
+  # "." installs the package in the current directory, which will be the
+  # location where the formula's tarball was unpacked.
+  venv.link_scripts(bin) { venv.pip_install "." }
```

Comment:
It seems redundant to have to call pip install twice and awkward to have the most import step buried inside a block passed to link_scripts.